mkosi/config.py

```diff
@@ -4618,7 +4627,8 @@ def setting_prohibited(self, setting: ConfigSetting[T]) -> bool:
         return (
             (not setting.tools and image == "tools")
             or (
-                setting.scope in (SettingScope.main, SettingScope.tools, SettingScope.multiversal)
+                setting.scope
+                in (SettingScope.main, SettingScope.tools, SettingScope.initrd, SettingScope.multiversal)
```

Comment:
This grouping could do with a name, maybe
```suggestion
                setting.scope.can_only_configured_on_main()
```